### DISCUSSION TITLE
Default value for BooleanGroup

### DISCUSSION THREAD

**Discussion by dgoldstein89** [laravel/nova-issues#3277] [Q&A] [answered]
Is there a way to set the default value for BooleanGroup? I tried returning an array like the following, but it doesn't seem to work...

```
BooleanGroup::make('Boolean Group')
        ->options(['one'=> 'One', 'two' => 'Two', 'three' => 'Three'])
        ->default(function () {
                return ['one'=> true, 'two' => false, 'three' => true];
        })
```

**Comment by dgoldstein89** [comment, marked as answer]
Update: It does work, mistake on my part.

**Reply by emjayess** [reply]
Is there not a way to provide the checkbox value/checked state right within the options? Obviously these options would need to be loaded from a data source and presented, so it seems fairly obtuse if the means to easily provide the state is lacking. 

And the nova documentation is sorely lacking in this regard.

**Reply by emjayess** [reply]
The `default` callback is never invoked on an edit/update form... is that by design?

**Comment by tigusigalpa** [comment]
Yes, you can make default value/checked for `BooleanGroup` Laravel Nova Field, like this:

`BooleanGroup::make('name')->options(['foo1' => 'bar1', 'foo2' => 'bar2'])->withMeta(['value' => ['foo1' => true]]) //if you want foo1 to be checked by default`
